drv/transceivers-server/src/main.rs

```diff
@@ -160,16 +190,117 @@ impl idl::InOrderTransceiversImpl for ServerImpl {
     }
 }
 
+impl NotificationHandler for ServerImpl {
+    fn current_notification_mask(&self) -> u32 {
+        TIMER_NOTIFICATION_MASK
+    }
+
+    // We currently only have one notification source so we are ignoring _bits
+    fn handle_notification(&mut self, _bits: u32) {
+        // Check for errors
+        let errors = self.leds.error_summary().unwrap();
+        if errors != self.led_error {
+            self.led_error = errors;
+            ringbuf_entry!(Trace::LEDErrorUpdate(errors))
+        }
+
+        // Query module presence and update LEDs accordingly
+        let presence = match self.transceivers.get_modules_status() {
+            Ok(status) => status.present,
+            Err(_) => 0,
+        };
+
+        if presence != self.modules_present {
+            // Errors are being suppressed here due to a miswiring of the I2C bus at
+            // the LED controller parts. They will not be accessible without rework
+            // to older hardware, and newer (correct) hardware will be replacing the
+            // hold stuff very soon.
+            // TODO: remove conditional compilation path once sidecar-a is sunset
+            cfg_if::cfg_if! {
+                if #[cfg(target_board = "sidecar-a")] {
+                    let _ = self.leds.update_led_state(presence);
+                } else {
+                    self.leds.update_led_state(presence).unwrap();
```

Comment:
I'm not sure if panicking the task is the right call when communication fails.  Maybe logging an error in the ringbuf instead, unless there's a good reason to restart the whole task?

Reply:
Good point. I agree that there is really no point in crashing the whole task due to problems with the LEDs. It will be good if we can recognize that an error has occurred in order to inform an operator (if necessary) that LEDs may not be functional, but the transceivers themselves may still be operational.